src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -3462,37 +3449,30 @@ void SimAirLoopComponent(EnergyPlusData &state,
         ManageOutsideAirSystem(state, CompName, FirstHVACIteration, AirLoopNum, CompIndex);
         // Fan Types for the air sys simulation
     } break;
-    case CompType::Fan_Simple_CV: { // 'Fan:ConstantVolume'
-        Fans::SimulateFanComponents(state, CompName, FirstHVACIteration, CompIndex);
-    } break;
-    case CompType::Fan_Simple_VAV: { // 'Fan:VariableVolume'
-        Fans::SimulateFanComponents(state, CompName, FirstHVACIteration, CompIndex);
+    case CompType::Fan_Simple_CV:  // 'Fan:ConstantVolume'
+    case CompType::Fan_Simple_VAV: // 'Fan:VariableVolume'
+    case CompType::Fan_ComponentModel: { // 'Fan:ComponentModel'
+        if (CompIndex == 0) {
+                CompIndex = Fans::GetFanIndex(state, CompName); // TODO: get rid of this
+        }
+        state.dataFans->fans(CompIndex)->simulate(state, FirstHVACIteration);
     } break;
+
     case CompType::Fan_System_Object: {                                        // "Fan:SystemModel" new for V8.6
-        if (CompIndex == 0) {                                                  // 0 means has not been filled because of 1-based arrays in old fortran
-            CompIndex = HVACFan::getFanObjectVectorIndex(state, CompName) + 1; // + 1 for shift from zero-based vector to 1-based compIndex
-        }
```

Comment:
CompIndex no longer needs to be protected here? I would think this is needed.

Reply:
Thanks. There are a bunch of crashes on Linux that I cannot reproduce on Mac, and this may be the cause. Although why would I not be able to reproduce them on Mac? 🤷‍♂️

Reply:
There is a one-time-init in SimAirServingZones::InitAirLoops that loops through the branches and components and gets the fan index (lines 1901 - 1960). That index could be saved in PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompIndex to avoid having to check for CompIndex == 0 here and above.

Reply:
And this check will not have to be added here if PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompIndex is set up early.

Reply:
> There is a one-time-init in SimAirServingZones::InitAirLoops that loops through the branches and components and gets the fan index (lines 1901 - 1960). That index could be saved in PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompIndex to avoid having to check for CompIndex == 0 here and above.

@rraustad this mostly works except ... there is an unhandled case (SASV.cc : 1928). 

```cpp
if (FoundOASys) {
    if (branch.DuctType != HVAC::AirDuctType::Heating) {
        SupFanIndex = comp.CompIndex = Fans::GetFanIndex(state, comp.Name);
        supFanType = state.dataFans->fans(SupFanIndex)->type;
	goto EndOfAirLoop;
    }    
    // Unhandled case: there is no else here
} else {
    ....
}
```

What is the desired behavior there? Supply fan? Return fan?